docs/source/usage/grouping.rst

```diff
@@ -0,0 +1,317 @@
+Grouping
+========
+
+QueryMate supports grouping query results by any field, including date fields with configurable granularity and timezone support. This is useful for building dashboards, reports, and analytics views.
+
+Basic Grouping
+--------------
+
+To group results by a field, use the ``group_by`` parameter:
+
+.. code-block:: python
+
+    from querymate import Querymate
+
+    querymate = Querymate(
+        select=["id", "name", "status"],
+        group_by="status",
+        limit=10,
+    )
+    result = querymate.run_grouped(db, User)
+
+Query parameter:
+
+.. code-block:: text
+
+    /users?q={"select":["id","name","status"],"group_by":"status","limit":10}
+
+Date Grouping with Granularity
+------------------------------
+
+For date fields, you can specify a granularity to group by time periods.
+
+Supported Granularities
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+
+   * - Granularity
+     - Description
+     - Key Format Example
+   * - ``year``
+     - Group by year
+     - ``2024``
+   * - ``month``
+     - Group by month
+     - ``2024-01``
+   * - ``day``
+     - Group by day
+     - ``2024-01-15``
+   * - ``hour``
+     - Group by hour
+     - ``2024-01-15T10``
+   * - ``minute``
+     - Group by minute
+     - ``2024-01-15T10:30``
+
+Examples
+~~~~~~~~
+
+.. code-block:: python
+
+    # Group by month
+    querymate = Querymate(
+        select=["id", "title", "created_at"],
+        group_by={"field": "created_at", "granularity": "month"},
+        limit=10,
+    )
+    result = querymate.run_grouped(db, Post)
+
+Query parameters:
+
+.. code-block:: text
+
+    # Group by year
+    /posts?q={"group_by":{"field":"created_at","granularity":"year"}}
+
+    # Group by day
+    /posts?q={"group_by":{"field":"created_at","granularity":"day"}}
+
+    # Group by hour
+    /posts?q={"group_by":{"field":"created_at","granularity":"hour"}}
+
+Timezone Support
+----------------
+
+When grouping by date, you can apply a timezone offset to ensure grouping aligns with your users' local time.
+
+Numeric Offset
+~~~~~~~~~~~~~~
+
+Use ``tz_offset`` to specify the offset in hours from UTC:
```

Comment:
I suggest to split this things related to timezone in other PR.
And also, i would pass a timezone or function